### ISSUE TITLE
How to configure the transnetv2.gin and reproduce the F1 of 77.9 on the ClipShots test set?

### ISSUE THREAD

**Issue by wentaozhu** [soCzech/TransNetV2#23]
Hi Tomáš,

Your work helps me a lot in understanding your great paper! Thank you so much!

I download the ClipShots training and training transitions datasets, and process them according to [https://github.com/soCzech/TransNetV2/blob/master/training/consolidate_datasets.py](url) and [https://github.com/soCzech/TransNetV2/blob/master/training/create_dataset.py](url)

I download the ClipShots test dataset and process it accordingly.

I also download the IACC.3 dataset and process it with the type of "train" .

I add the ClipShots training, training transitions and IACC.3 in the "options.trn_files" of [https://github.com/soCzech/TransNetV2/blob/master/configs/transnetv2.gin](url), and add ClipShots test in the "options.tst_files". I also change "options.n_epochs" to 50 as indicated in the paper.

However, I can only obtain F1 of 0.74. Could you please give more training details and instructions on how to reproduce 77.9 on the test set?

What are the meanings of file names in "options.tst_files" and how to generate these files?

I also use the pretrained weights in [https://github.com/soCzech/TransNetV2/tree/master/inference/transnetv2-weights](url) to test the ClipShots test dataset by revising "options.restore" and "options.test_only" to True in [https://github.com/soCzech/TransNetV2/blob/master/configs/transnetv2.gin](url). I can only get F1 of 0.2545 and cannot reproduce 77.9.

I appreciate your great help so much!

Wentao

**Comment by ghost** [comment]
I have the same problem. BTW, how can you download the IACC.3 dataset? I find only iacc.3.master.shot.reference is publicly available. Dose the paper use this dataset to achieve 77.9?

**Comment by soCzech** [comment]
**Here is the response I gave to @wentaozhu by email:**

> However, I can only obtain F1 of 0.74. Could you please give more training details and instructions on how to reproduce 77.9 on the test set?

The test loop during training takes frames sequentially without overlap (https://github.com/soCzech/TransNetV2/blob/master/training/input_processing.py#L568 ). Eg. frame 1 to frame 100, then frame 101 to 200, etc.
This means the edge frames such as frame 0, even frames 1, 2, 3,... have no / almost no context from the left side. This obviously produces worse results than having bigger context for each frame. To get the same numbers you need to make some overlap between consequent batches. In the paper we put in eg. frames 1 to 100 but use predictions only for frames 25 to 75 and than in the next batch we put frames 75 to 175 and again use predictions for frames 100 to 150.
These calculations are done in https://github.com/soCzech/TransNetV2/blob/master/training/evaluate.py#L82 . The train script for simplicity and speed does not do this.

> What are the meanings of file names in "options.tst_files" and how to generate these files?

Well since you have tried to reproduce the network you had to use them, right? After each epoch, these files are used for approximate evaluation as mentioned above. They can by generated by https://github.com/soCzech/TransNetV2/blob/master/training/create_dataset.py#L86 .

> I also use the pretrained weights in https://github.com/soCzech/TransNetV2/tree/master/inference/transnetv2-weights to test the ClipShots test dataset by revising "options.restore" and "options.test_only" to True in https://github.com/soCzech/TransNetV2/blob/master/configs/transnetv2.gin. I can only get F1 of 0.2545 and cannot reproduce 77.9.

Again, don't use the train script to get final F1 score, use evaluate.py.
But 25% is extremely low even for the train.py evaluation, maybe there is some issue with the ground truth mapping? Maybe try visualizing the frames, the ground truth and the predictions to see where is the problem. For the debugging purposes you can also try evaluation on BBC dataset, which is still large but is much, much, much cleaner with no errors in the ground truth. The clip shots have videos in the test set which are not annotated (also the reason why the numbers are that low - meaning 77%, not 25:)).

**Comment by soCzech** [comment]
> How can you download the IACC.3 dataset?

You need to write to the authors, it is not publicly available due to copyrights.

> Does the paper use this dataset to achieve 77.9?

Yes, but there is probably nothing special about IACC.3. You can probably achieve similar results with diferent datasets.

**Comment by ghost** [comment]
Thanks for your response. I have tested the weights you published and achieve 77.89. Thanks for your great work!
